### PR TITLE
enable auto-detection of 29 bit OBDII. resolves #890

### LINKED ISSUE #890: 29 / 11 bit OBDII support
Some vehicles, like certain honda products only support 29 bit CAN OBDII. need to support this.

### PR DESCRIPTION
**Pull request by brentpicasso**
Enable auto-detection of 29 or 11 bit OBDII protocol.

### CODE REVIEW

File: include/CAN/can_mapping.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ CPP_GUARD_BEGIN
  * match the can message based on the specified CAN mapping ID and ID mask
  * @param can_msg the CAN message to test
  * @param mapping the can mapping to check against
- * @return true if the CAN message matches the mapping's CAN ID and ID mask
+ * @return true if the CAN message matches the mapping's CAN ID and ID mask, or true if mapping's CAN ID is 0
```
Comment (stieg) [inline include/CAN/can_mapping.h]: Why 0?  What is special about that?  Does this not break behavior of the function?

Comment (brentpicasso) [inline include/CAN/can_mapping.h]: I changed the API so that a CAN ID of 0 means "match any CAN ID".

Comment (brentpicasso) [inline include/CAN/can_mapping.h]: (like a wildcard)

Comment (brentpicasso) [inline include/CAN/can_mapping.h]: Ok, enhanced documentation.

**Pull request merged**